src/govuk/components/notification-banner/notification-banner.test.js

```diff
@@ -30,30 +33,104 @@ describe('/components/notification-banner/with-type-as-success', () => {
-  describe('when auto-focus is disabled', () => {
-    it('does not have a tabindex attribute', async () => {
+  describe('and auto-focus is disabled using data attributes', () => {
+    beforeAll(async () => {
       await page.goto(`${baseUrl}/components/notification-banner/auto-focus-disabled,-with-type-as-success/preview`, { waitUntil: 'load' })
+    })
 
+    it('does not have a tabindex attribute', async () => {
       const tabindex = await page.$eval('.govuk-notification-banner', el => el.getAttribute('tabindex'))
 
       expect(tabindex).toBeNull()
     })
 
     it('does not focus the notification banner', async () => {
-      await page.goto(`${baseUrl}/components/notification-banner/auto-focus-disabled,-with-type-as-success/preview`, { waitUntil: 'load' })
+      const activeElement = await page.evaluate(() => document.activeElement.dataset.module)
+
+      expect(activeElement).not.toBe('govuk-notification-banner')
+    })
+  })
+
+  describe('and auto-focus is disabled using JavaScript configuration', () => {
+    beforeAll(async () => {
+      await renderAndInitialise('notification-banner', {
+        baseUrl,
+        nunjucksParams:
+          examples['with type as success'],
+        javascriptConfig: {
+          disableAutoFocus: true
+        }
+      })
+    })
+
+    it('does not have a tabindex attribute', async () => {
+      const tabindex = await page.$eval('.govuk-notification-banner', el => el.getAttribute('tabindex'))
+
+      expect(tabindex).toBeNull()
+    })
 
+    it('does not focus the notification banner', async () => {
+      const activeElement = await page.evaluate(() => document.activeElement.dataset.module)
+
+      expect(activeElement).not.toBe('govuk-notification-banner')
+    })
+  })
+
+  describe('and auto-focus is disabled using options passed to initAll', () => {
+    beforeAll(async () => {
+      await renderAndInitialise('notification-banner', {
+        baseUrl,
+        nunjucksParams:
+          examples['with type as success'],
+        initialiser () {
+          window.GOVUKFrontend.initAll({
+            notificationBanner: {
+              disableAutoFocus: true
+            }
+          })
+        }
+      })
+    })
+
+    it('does not have a tabindex attribute', async () => {
+      const tabindex = await page.$eval('.govuk-notification-banner', el => el.getAttribute('tabindex'))
+
+      expect(tabindex).toBeNull()
+    })
+
+    it('does not focus the notification banner', async () => {
       const activeElement = await page.evaluate(() => document.activeElement.dataset.module)
 
       expect(activeElement).not.toBe('govuk-notification-banner')
     })
   })
-})
 
-describe('components/notification-banner/role=alert-overridden-to-role=region,-with-type-as-success', () => {
-  describe('when role is not alert', () => {
+  describe('and autofocus is disabled in JS but enabled in data attributes, attributes win', () => {
```

Comment:
"attributes win" probably makes this suite easier to parse, but semantically probably belongs in the tests within the description?

Like, `it('is automatically focused when the page loads, as per data attributes')` or something.

I'm not particularly fussed either way.